background.js

```diff
@@ -49,47 +91,45 @@ const waitAndClick = async (querySelector) => {
   if (await waitForElement(querySelector)) await clickElement(querySelector);
 };
 
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
+browser.pageAction.onClicked.addListener(async (tab) => {
+  browser.tabs.executeScript({
     code: `
-    if (!document.querySelector(".caal-alerts")) {
-      alerts = document.createElement("div")
-      alerts.setAttribute("class", "caal-alerts");
-      alerts.setAttribute("style", "position:absolute;");
-      document.body.appendChild(alerts);
-    }
-  `,
+      if (!document.querySelector(".caal-loading")) {
+        const loading = document.createElement("div");
+        loading.textContent = "Analysing chess game...";
+        loading.setAttribute("class", "caal-loading");
+        document.body.appendChild(loading);
+      }
+    `,
   });
 
-  await waitAndClick('.share');
+  await setLoadingState(true);
 
-  await waitAndClick('.board-tab-item-underlined-component');
+  const [playerName] = await browser.tabs.executeScript({
+    code: `document.querySelector('[data-test-element="user-tagline-username"]').textContent;`,
+  });
 
-  if (await waitForElement("[name='pgn']")) {
-    const [pgn] = await browser.tabs.executeScript({
-      code: `document.querySelector("[name='pgn']").value;`,
-    });
+  const pgn = await getPGN(playerName, tab.url.split("?")[0]);
```

Comment:
What if the `pgn` cannot be found and the function returns `null`?

Reply:
`null` value for `pgn` is now properly handled